test/test_post.f90

```diff
@@ -0,0 +1,54 @@
+program test_post
+    use iso_fortran_env, only: stderr => error_unit
+    use http, only: request, header_type, HTTP_POST, response_type
+    implicit none
+    type(response_type) :: res
+    character(:), allocatable :: json_data, original_content, msg
+    type(header_type), allocatable :: req_header(:)
+    logical :: ok = .true.
+
+    original_content = '{"id":101,"title":"BMW","description":"A luxurious and high-performance vehicle"}'
+    req_header = [header_type('Content-Type', 'application/json')]
+
+    json_data = '{"title":"BMW","description":"A luxurious and high-performance vehicle"}'
+
+    res = request(url='https://dummyjson.com/products/add', method=HTTP_POST, data=json_data, header=req_header)
```

Comment:
Why do we introduce all these different mock services into tests and examples? Can't httpbin.org cover all our needs? Let's do some research on which one service (if any) can cover all our needs and use just that one.

Reply:
There are two reasons why httpbin is not used:

1. Sometimes, when making a post request, httpbin returns a "504 Gateway Time-out" error.
2. The response content string from httpbin is not delivered in a single line, i.e. response content comes in a different line separated by a new line character. which can make it difficult to compare two strings. because I think Fortran, may not support the use of newline characters in a string to store.